barcode-scanner-service/barcode-scanner.py

```diff
@@ -0,0 +1,180 @@
+        24:['u', 'U'],
+        25:['v', 'V'],
+        26:['w', 'W'],
+        27:['x', 'X'],
+        28:['y', 'Y'],
+        29:['z', 'Z'],
+        30:['1', '!'],
+        31:['2', '@'],
+        32:['3', '#'],
+        33:['4', '$'],
+        34:['5', '%'],
+        35:['6', '^'],
+        36:['7' ,'&'],
+        37:['8', '*'],
+        38:['9', '('],
+        39:['0', ')'],
+        40:['\n', '\n'],
+        41:['\x1b', '\x1b'],
+        42:['\b', '\b'],
+        43:['\t', '\t'],
+        44:[' ', ' '],
+        45:['_', '_'],
+        46:['=', '+'],
+        47:['[', '{'],
+        48:[']', '}'],
+        49:['\\', '|'],
+        50:['#', '~'],
+        51:[';', ':'],
+        52:["'", '"'],
+        53:['`', '~'],
+        54:[',', '<'],
+        55:['.', '>'],
+        56:['/', '?'],
+        100:['\\', '|'],
+        103:['=', '='],
+        }
+
+    # A 2 in first byte seems to indicate to shift the key. For example
+    # a code for ';' but with 2 in first byte really means ':'.
+    if lst[0] == 2:
+        shift = 1
+    else:
+        shift = 0
+        
+    # The character to convert is in the third byte
+    ch = lst[2]
+    if ch not in conv_table:
+        print("Warning: data not in conversion table")
+        return ''
+    return conv_table[ch][shift]
+
+
+
+parser = argparse.ArgumentParser(description='Sends requests via KServe gRPC API using images in format supported by OpenCV. It displays performance statistics and optionally the model accuracy')
+parser.add_argument('--vid', required=False, default=0x05e0, help='Vendor ID of the barcode scanner') # Default value for Symbol LS2208 General Purpose Barcode Scanner
+parser.add_argument('--pid', required=False, default=0x1200, help='Product ID of the barcode scanner') # Default value for Symbol LS2208 General Purpose Barcode Scanner
+parser.add_argument('--mqtt_host', required=False, default="localhost", help='Product ID of the barcode scanner')
+args = vars(parser.parse_args())
+
+# Find our device using the VID (Vendor ID) and PID (Product ID)
+dev = usb.core.find(idVendor=args['vid'], idProduct=args['pid'])
+if dev is None:
+    raise ValueError('USB device not found')
+
+# Disconnect it from kernel
+needs_reattach = False
+if dev.is_kernel_driver_active(0):
+    needs_reattach = True
+    dev.detach_kernel_driver(0)
+    print("Detached USB device from kernel driver")
+
+# set the active configuration. With no arguments, the first
+# configuration will be the active one
+dev.set_configuration()
+ 
+# get an endpoint instance
+cfg = dev.get_active_configuration()
+intf = cfg[(0,0)]
+
+ep = usb.util.find_descriptor(
+    intf,
+    # match the first IN endpoint
+    custom_match = \
+    lambda e: \
+        usb.util.endpoint_direction(e.bEndpointAddress) == \
+        usb.util.ENDPOINT_IN)
+
+assert ep is not None, "Endpoint for USB device not found. Something is wrong."
+
+mqttClient = paho.Client()
+mqttClient.connect(args['mqtt_host'], 1883, 60)
```

Comment:
do we want to be able to pass in our MQTT parameters? should this be something we pass to the arg parser in case we have a secure broker?